packages/flutter/test/services/asset_bundle_test.dart

```diff
@@ -178,4 +188,11 @@ void main() {
     expect(assetManifest.keys.toList(), equals(<String>['one']));
     expect(assetManifest['one'], <Object>[]);
   });
+
+  test('loadStructuredBinaryData cache is populated synchronously if load is synchronous', () {
+    final SynchronousTestAssetBundle bundle = SynchronousTestAssetBundle();
+    bundle.loadStructuredBinaryData('one', (ByteData data) => 1);
+    final FutureOr<int> data = bundle.loadStructuredBinaryData('one', (ByteData data) => 0);
+    expect(data, isA<SynchronousFuture<int>>());
+  });
```

Comment:
How does this fail if your change isn't present?

Reply:
It totally didn't. My bad. Should be fixed now.